stores/sql/database.go

```diff
@@ -27,13 +27,19 @@ type (
 		DeleteObject(ctx context.Context, bucket, key string) (bool, error)
 
 		// DeleteObjects deletes a batch of objects starting with the given
-		// prefix and returns 'true' if any object was deleted. The deletion
-		// should prioritize larger objects for a more predictable performance.
+		// prefix and returns 'true' if any object was deleted.
 		DeleteObjects(ctx context.Context, bucket, prefix string, limit int64) (bool, error)
 
 		// MakeDirsForPath creates all directories for a given object's path.
 		MakeDirsForPath(ctx context.Context, path string) (uint, error)
 
+		// PruneDirs prunes any directories that are empty.
+		PruneDirs(ctx context.Context) error
```

Comment:
maybe we should rename this to `PruneEmptyDirs` and while we're at it rename `PruneSlabs` to `PruneOrphanSlabs` to make it immediately apparent what we are doing